plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/ProxySettings.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3;
+
+import com.amazonaws.Protocol;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.SettingsException;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.Objects;
+
+public class ProxySettings {
+    public static final ProxySettings NO_PROXY_SETTINGS = new ProxySettings(ProxyType.DIRECT, null, -1, null, null);
+
+    public static enum ProxyType {
+        HTTP(Protocol.HTTP.name()),
+        HTTPS(Protocol.HTTPS.name()),
+        SOCKS("SOCKS"),
```

Comment:
Do we actually need to introduce `SOCKS` proxy type? Shouldn't we treat it as any other unsupported value?

Reply:
It is possible to support SOCKS. Socks woks on the Socket level. So extending SSLSocketFactory is ok IMHO

Reply:
Nevermind, I got that, conversion to protocol is not supported for SOCKS